spec.html

```diff
@@ -787,68 +787,79 @@ <h1>Throw an Exception</h1>
+        <p>is equivalent to the sequence of steps:</p>
         <emu-alg>
-          1. Let _hygienicTemp_ be AbstractOperation().
+          1. Let _hygienicTemp_ be *[argument]*.
           1. If _hygienicTemp_ is an abrupt completion, return _hygienicTemp_.
           1. Else if _hygienicTemp_ is a Completion Record, set _hygienicTemp_ to _hygienicTemp_.[[Value]].
+          1. *[before]* _hygienicTemp_ *[after]*
         </emu-alg>
-        <p>Where _hygienicTemp_ is ephemeral and visible only in the steps pertaining to ReturnIfAbrupt.</p>
-        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <p>where</p>
+        <ul>
+          <li>*[before]*, if present, is any algorithm text that does not contain ReturnIfAbrupt.</li>
+          <li>*[after]*, if present, is any algorithm text.</li>
+          <li>*[argument]* is an alias, an application of an abstract operation, an application of a syntax-directed operation, or a use of ReturnIfAbrupt.</li>
+          <li>_hygienicTemp_ is ephemeral and visible only in the steps pertaining to this ReturnIfAbrupt expansion.</li>
+        </ul>
+
+        <p>For example, an algorithm step that says:</p>
         <emu-alg>
-          1. Let _result_ be AbstractOperation(ReturnIfAbrupt(_argument_)).
+          1. Set _someAlias_ to _someValue_ if ReturnIfAbrupt(_someValue_.OperationName(_firstArgument_, _secondArgument_)) is *true*.
         </emu-alg>
-        <p>mean the same thing as:</p>
+        <p>is equivalent to the sequence of steps:</p>
         <emu-alg>
-          1. If _argument_ is an abrupt completion, return _argument_.
-          1. If _argument_ is a Completion Record, set _argument_ to _argument_.[[Value]].
-          1. Let _result_ be AbstractOperation(_argument_).
+          1. Let _hygienicTemp_ be _someValue_.OperationName(_firstArgument_, _secondArgument_).
+          1. If _hygienicTemp_ is an abrupt completion, return _hygienicTemp_.
+          1. Else if _hygienicTemp_ is a Completion Record, set _hygienicTemp_ to _hygienicTemp_.[[Value]].
+          1. Set _someAlias_ to _someValue_ if _hygienicTemp_ is *true*.
         </emu-alg>
+
+        <p>Note that the resulting steps of ReturnIfAbrupt expansion may themselves contain ReturnIfAbrupt. In these cases, expansion can be applied to the resulting steps until the steps no longer contain ReturnIfAbrupt.</p>
       </emu-clause>
+
       <emu-clause id="sec-returnifabrupt-shorthands">
-        <h1>ReturnIfAbrupt Shorthands</h1>
-        <p>Invocations of abstract operations and syntax-directed operations that are prefixed by `?` indicate that ReturnIfAbrupt should be applied to the resulting Completion Record. For example, the step:</p>
+        <h1>Shorthands Relating to Completion Records</h1>
+
+        <p>Invocations of abstract operations may be prefixed by `?` as shorthand for applying the ReturnIfAbrupt expansion.</p>
         <emu-alg>
           1. ? OperationName().
```

Comment:
not sure if we need to call out `Perform` (ie, `Perform ? Op()` or `Perform ! Op()`), or if that's a topic for a different PR.